staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -623,10 +652,62 @@ func preprocessDeletionListForMerging(key string, original map[string]interface{
 	return false, false, "", nil
 }
 
+// processReplaceKeys processes replaceKeys list.
+// When MergeParallelList is true, we validate the patch and
+// then clear the fields that are not in the list.
+func processReplaceKeys(original, patch map[string]interface{}, options MergeOptions) error {
+	replaceKeysInPatch, foundInPatch := patch[replaceKeysDirective]
+	delete(patch, replaceKeysDirective)
+	if !options.MergeParallelList {
+		replaceKeysInOriginal, foundInOriginal := original[replaceKeysDirective]
+		switch {
+		case foundInOriginal && foundInPatch:
+			if !reflect.DeepEqual(replaceKeysInOriginal, replaceKeysInPatch) {
+				return fmt.Errorf("%v and %v are not deep equal", replaceKeysInOriginal, replaceKeysInPatch)
+			}
+		case !foundInOriginal && foundInPatch:
```

Comment:
Agree.
I thought I would have 4 cases here, but I actually got 2. And then I forgot it.
I will update it soon.

> I think there are even more simplifications possible at the beginning of this function.

Feel free to point out. Review welcome.